### ISSUE TITLE
Sass Syntax - Tests progress

### ISSUE THREAD

**Issue by bgriffith** [sasstools/sass-lint#166]
Having had a few Sass Syntax related issues popping up, I think it would be a good move to create tests for that syntax too.

I've started going through the current tests and have created quite a few .sass equivalents. So far there has been a few tests that have failed but the majority are ok.

By testing both the SCSS and Sass syntax I feel we will then be able to confidently say that we have cross syntax support.

I'll update the list below as I go.

### The rules

- [x] border-zero
- [x] clean-import-paths - (Issue #179, Fixed #205)
- [x] empty-args
- [x] empty-line-between-blocks - (:bug: Issue #216)
- [x] extends-before-declarations
- [x] extends-before-mixins - (Fixed issue #203)
- [x] final-newline - (Issue #207, Fixed #208)
- [x] hex-length
- [x] hex-notation
- [x] indentation
- [x] leading-zero
- [x] mixing-before-declarations
- [x] nesting-depth
- [x] no-color-keywords
- [x] no-color-literals (Issue #186, Fixed in #187)
- [x] no-css-comments
- [x] no-debug
- [x] no-duplicate-properties
- [x] no-extends - (Fixed issue - #173)
- [x] no-ids
- [x] no-important
- [x] no-invalid-hex
- [x] no-qualifying-elements
- [x] no-trailing-zeros
- [x] no-url-protocols
- [x] no-vendor-prefix (Issue #182, Fixed in #183)
- [x] no-warn
- [x] placeholder-in-extend - (Fixed in #204)
- [x] property-sort-order
- [x] quotes
- [x] shorthand values
- [x] single-line-per-selector - (Fixed issue - #165)
- [x] space-after-bang
- [x] space-after-colon
- [x] space-after-comma
- [x] space-before bang
- [x] space-before-colon
- [x] space-before-parens
- [x] url-quotes
- [x] variable-for-property
- [x] zero-unit

### Rules that are N/A
- trailing-semicolon
- space-before-brace
- no-empty-rulesets
- brace-style
- one-declaration-per-line

**Comment by DanPurdy** [comment]
Awesome, I started having a look at this after we spoke this morning, but looks like you've got further ahead than me. Do you wanna explain a little how you're going about it, I've got a few rules nearly done here that i'd like to get the sass tests up and running for too, be good to have them in the same format as yours.

**Comment by bgriffith** [comment]
Sure, it's very simple and happy to change if you've got a better suggestion.

All I'm doing is creating .sass versions of the .scss files (with any adjustments if required) and updating the mocha test file to test for both.

Ideally we should be writing the actual tests in a way that's syntax agnostic so in my mind we shouldn't have two separate tests but simply improve the current one's to take into account any issues (currently not too many).

```js
'use strict';

var lint = require('./_lint');

describe('no extends', function () {
  it('scss - enforce', function (done) {
    var file = lint.file('no-extends.scss');

    lint.test(file, {
      'no-extends': 1
    }, function (data) {
      lint.assert.equal(1, data.warningCount);
      done();
    });
  });

  it('sass - enforce', function (done) {
    var file = lint.file('no-extends.sass');

    lint.test(file, {
      'no-extends': 1
    }, function (data) {
      lint.assert.equal(1, data.warningCount);
      done();
    });
  });
});
```

**Comment by DanPurdy** [comment]
This will be ok to start with but I think the duplication will become a little unmanageable for the rules where there are lots of test cases. 

The tests should give the same results for both scss and sass, but I guess seen as some of the rules won't apply to the sass side of things it adds a small challenge there, for now something as simple as setting the `file` variable to an array of the 2 different syntaxes and then just looping the tests could work for now?

**Comment by DanPurdy** [comment]
Followed your lead I believe for this new rule, we can revisit once everything is up and running again?

**Comment by DanPurdy** [comment]
You should look to get this up as a PR soon, we can then keep track a bit better of what still needs to be done and any issues we find in separate issues. Also it'll give us a nice base from which to get sass tests as the norm for any further rules being written.

**Comment by DanPurdy** [comment]
Going to wait for the latest version of `gonzales-pe` to rewrite the indentation. Moving this out of 1.4.0 for now.

**Comment by bgriffith** [comment]
Indentation is now good.
